### PR TITLE
Localization : Site settings supported culture

### PR DESCRIPTION
**Pull request by Skrypt**
Add SiteExtension method GetManagableCultures()

Fixes https://github.com/OrchardCMS/OrchardCore/issues/3646

### CODE REVIEW

**Comment by hishamco** [comment]
I dislike the change for a reason, why you didn't change the value of the `site.SupportedCultures` itself instead of changing anyone who called it .. this give us a consistency everywhere across, if you don't mind I can push some commits to your branch

**Comment by Skrypt** [comment]
No, I won't change that value as this would cause other issues elsewhere. I tried and it messes up the admin UI where we add/remove those values.

**Comment by hishamco** [comment]
Of course it causes issues right now, because the main issue from the beginning. But if we fix it will guarantee that's no else in future will messes up

> And for consistency then you can use GetManageableCultures() now.

Got it .. but still extra code and this is really confusing for new comers

**Comment by hishamco** [comment]
Just one question and you can ignore my previous comment, is this mean that `site.SupportedCultures` is useless now?!!

**Comment by Skrypt** [comment]
No it's still used but it's not appropriate to use it for determining which culture is currently supported by the Localization Middleware. site.SupportedCulture is a IEntity which means it's used for storing the settings in the database. site.SupportedCulture.Concat(site.Culture).Distinct.ToArray() is what the Localization Middleware is using. Though, the Localization Middleware is not always enabled so for the admin side of things it is more appropriate to fetch the manageable cultures by using the extension method we created @Skrypt @jtkech

**Comment by hishamco** [comment]
So that means the only case I used that property when I add the supported culture from the admin, nothing else

**Comment by hishamco** [comment]
@Skrypt If the team accept the PR I have few changes I need to add, so how can I add commits to your PR?

**Comment by Skrypt** [comment]
We will only change the name of the method. It should be merged soon so if you want to add things that uses it or extend it then create a branch starting with https://github.com/OrchardCMS/OrchardCore/tree/skrypt/%233646 or simply https://github.com/OrchardCMS/OrchardCore when it will get merged.

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System.Linq;
+
+namespace OrchardCore.Settings
+{
+    public static class SiteExtensions
+    {
+        public static string[] GetManageableCultures(this ISite site) {
```
Comment (jtkech) [inline src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteExtensions.cs]: Formatting

**Comment by hishamco** [comment]
I checkout your branch but I got access denied when I push my changes

If I create a new branch starting with your it will be a duplicate. Am I right?

**Comment by sebastienros** [comment]
@hishamco what do you want to change?

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteExtensions.cs (hunk 1 above)
Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteExtensions.cs]: done

**Comment by Skrypt** [comment]
Here are some name ideas : 

1. - [ ] GetConfiguredCultures
2. - [ ] GetSettedCultures
3. - [ ] GetSelectableCultures
4. - [ ] GetSelectedCultures

**Comment by hishamco** [comment]
> @hishamco what do you want to change?

I request one change, also we could apply this to localisation module too

**Comment by Skrypt** [comment]
Done

**Comment by Skrypt** [comment]
Renamed it to GetConfiguredCultures since it makes more sense than GetManageableCultures. Though still open for changing it.

**Comment by Skrypt** [comment]
GetConfiguredSupportedCultures ?

**Comment by Skrypt** [comment]
@hishamco you need to create a fork of Orchard Core under your Github account and then you can create branch in your fork and send pull requests from there. I assume that it is what you are doing normally. You could simply update your fork so that you get all the latest changes including my branch.

**Comment by hishamco** [comment]
I got your branch, and I add my commits, but I got access denied when I push because I'm not a member of OrchardCore organization

**Comment by Skrypt** [comment]
That's because you can't do that. Normally you create a PR that targets my branch. I added your changes already so no need now. Thanks for the suggestions they we're right.

**Comment by hishamco** [comment]
There's a need because you miss one of my changes 😊

**Pull request merged**